File: backend/Models/Book.cs
using Microsoft.AspNetCore.Http.HttpResults;
using System.ComponentModel.DataAnnotations.Schema;
using System.Net;

namespace Chapter.Models
{
    [Table("Books")]
    public class Book
    {
        public string Id { get;  set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public string Authors { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string ThumbnailUrl { get; set; } = string.Empty;
        public string Publisher { get; set; } = string.Empty;
        public DateOnly? PublishedDate { get; set; }
        public int? PageCount { get; set; }
       

        public virtual ICollection<UserLibrary> LibraryEntries { get; set; } = new List<UserLibrary>();
        public virtual ICollection<Rating> Ratings { get; set; } = new List<Rating>();
        public virtual ICollection<Review> Reviews { get; set; } = new List<Review>();


    }
}

File: frontend/types/book.ts
export type Book = {
  volumeInfo: {
    imageLinks?: {
      thumbnail: string
    }
    title: string
    authors: string[]
  }
  id: string
}

export type BackendBook = {
  title: string
  authors: string
  description: string
  thumbnailUrl: string
  publisher: string
  publishedDate: string
  libraryEntries: string[]
  pageCount: number
}

export type Review = {
  id: number
  title: string
  content: string
  createdAt: string
  updatedAt: string
  createdBy: string
}

export type CollectionItem = {
  id: number
  status: string
  addedDate: string
  bookId: string
  title: string
  authors: string
  thumbnailUrl: string
}
